opm/parser/eclipse/EclipseState/IOConfig/IOConfig.cpp

```diff
@@ -0,0 +1,238 @@
+        rs.basic     = 0;
+        rs.frequency = 1;
+
+        m_timemap = timemap;
+        m_restart_output_config = std::make_shared<DynamicState<restartConfig>>(timemap, rs);
+    }
+
+    void IOConfig::handleSolutionSection(std::shared_ptr<const SOLUTIONSection> solutionSection) {
+
+        if (m_timemap && solutionSection->hasKeyword("RPTRST")) {
+            auto rptrstkeyword = solutionSection->getLastKeyword("RPTRST");
+            size_t currentStep = 0;
+
+            DeckRecordConstPtr record = rptrstkeyword->getRecord(0);
+
+            size_t basic = 1;
+            size_t freq  = 0;
+
+            DeckItemConstPtr item = record->getItem(0);
+
+            for (size_t index = 0; index < item->size(); ++index) {
+
+                if (item->hasValue(index)) {
+                    std::string mnemonics = item->getString(index);
+                    std::size_t found_basic = mnemonics.find("BASIC=");
+                    if (found_basic != std::string::npos) {
+                        std::string basic_no = mnemonics.substr(found_basic+6, found_basic+7);
+                        basic = atoi(basic_no.c_str());
+                    }
+
+                    std::size_t found_freq = mnemonics.find("FREQ=");
+                    if (found_freq != std::string::npos) {
+                        std::string freq_no = mnemonics.substr(found_freq+5, found_freq+6);
+                        freq = atoi(freq_no.c_str());
+                    }
+                }
+            }
+            handleRPTRSTBasic(m_timemap, currentStep, basic, freq, true);
+        }
+    }
+
+
+
+    void IOConfig::handleGridSection(std::shared_ptr<const GRIDSection> gridSection) {
+        m_write_INIT_file = gridSection->hasKeyword("INIT");
+
+        if (gridSection->hasKeyword("GRIDFILE")) {
+            auto gridfilekeyword = gridSection->getKeyword("GRIDFILE");
+            if (gridfilekeyword->size() > 0) {
+                auto rec = gridfilekeyword->getRecord(0);
+                auto item1 = rec->getItem(0);
+                if ((item1->hasValue(0)) && (item1->getInt(0) !=  0)) {
+                    throw std::runtime_error("IOConfig: Reading GRIDFILE keyword from GRID section: Output of GRID file is not supported") ;
+                }
+                if (rec->size() > 1) {
+                    auto item2 = rec->getItem(1);
+                    if ((item2->hasValue(0)) && (item2->getInt(0) ==  0)) {
+                        m_write_EGRID_file = false;
+                    }
+                }
+            }
+        }
+        if (gridSection->hasKeyword("NOGGF")) {
+            m_write_EGRID_file = false;
+        }
+    }
+
+
+    void IOConfig::handleRunspecSection(std::shared_ptr<const RUNSPECSection> runspecSection) {
+        m_FMTOUT  = runspecSection->hasKeyword("FMTOUT");  //Output files are formatted
+        m_UNIFOUT = runspecSection->hasKeyword("UNIFOUT"); //Output files are to be unified
+    }
+
+
+    bool IOConfig::getUNIFOUT() const {
```

Comment:
Add support for the `UNIFIN` keyword as well - that is 100% analogous to the `UNIFOUT` keyword.